### PR TITLE
fix(CLDX-128): update tasks as per testing of push-to-dev-portal pipeline

### PR DESCRIPTION
**Pull request by pkhander**
This commit fixes bugs for the tasks required for push-binaries-to-dev-portal after testing

### CODE REVIEW

File: tasks/publish-to-cgw/publish-to-cgw.yaml [hunk 1]
```diff
@@ -70,9 +70,10 @@ spec:
         import subprocess
 
         DATA_FILE = "$(workspaces.data.path)/$(params.dataPath)"
+        CONTENT_DIR = "$(workspaces.data.path)/$(params.contentDir)"
```
Comment (johnbieren) [inline tasks/publish-to-cgw/publish-to-cgw.yaml:73]: Why this change? It makes the task no longer safe against overwrites if parallel pipelineruns are running

Comment (pkhander) [reply tasks/publish-to-cgw/publish-to-cgw.yaml:73]: Previously we had the following:
```
WORKSPACE_DIR = "/workspace/data/{sub_dir}"
CONTENT_DIR = f"{WORKSPACE_DIR}/$(params.contentDir)" 
```
$(params.contentDir) results in "{sub_dir}/binaries"
This change is made so we don't have the extra {sub_dir} passed from extract-binaries-from-image

Comment (johnbieren) [reply tasks/publish-to-cgw/publish-to-cgw.yaml:73]: The code I see is 
```
WORKSPACE_DIR = os.path.dirname(DATA_FILE)
```
not `WORKSPACE_DIR = "/workspace/data/{sub_dir}"`
I am the one who asked for the WORKSPACE_DIR to be set based on the dirname of the data file. Collect-data puts all the jsons in a subdir. So, using the same dir as the data file ensures you are in a dir specific to this pipelinerun. The only reason why things work now is because in the pipeline definition we pass the same value to subdir for the extract-binaries task as we do to collect-data, but IMO it isn't obvious they *must* be the same. 
https://issues.redhat.com/browse/RELEASE-1003 is open to address it. Am I making sense? With the proposed changes here, you can collide. All it takes is $(workspaces.data.path) to be the same for two pipelineruns (which if you have one volume, it will be) and then for params.contentDir to be the same for two pipelineruns

Comment (pkhander) [reply tasks/publish-to-cgw/publish-to-cgw.yaml:73]: I mean to say that `WORKSPACE_DIR = os.path.dirname(DATA_FILE)` translates to `WORKSPACE_DIR = "/workspace/data/{sub_dir}"` 

I agree dirname approach is best.
The reason for this small change is that we were ending up with `workspace/data/sub_dir/sub_dir/binaries`
notice the duplicate sub_dir

I see two options to make this work.

1.  extract-binaries-to-image results just binaries instead of subdir/binaries
2.  get basename of  `$(params.contentDir)` and then append it to  WORKSPACE_DIR

Comment (johnbieren) [reply tasks/publish-to-cgw/publish-to-cgw.yaml:73]: So are you saying `$(params.contentDir)` already contains the pipelinerun specific subdir?

Comment (pkhander) [reply tasks/publish-to-cgw/publish-to-cgw.yaml:73]: yea, correct.

**Review by johnbieren — DISMISSED** [review]

**Review by johnbieren — APPROVED** [review]
We added a new shellchecker to the repo, which you are seeing alerts for
